### PR TITLE
remove default enable ACPI kconfig macro such as DSDT and ACPI_SHELL

### PR DESCRIPTION
**Pull request by najumon1980**
remove default enable kconfig macro such as DSDT and ACPI_SHELL if PCIE or SHELL enabled. All Intel platform may not require to use ACPI DSDT related APIs and hence it should not enable default if PCIE enabled.

### CODE REVIEW

File: boards/x86/intel_adl/intel_adl_crb_defconfig [hunk 1]
```diff
@@ -13,3 +13,4 @@ CONFIG_SMP=y
 CONFIG_BUILD_OUTPUT_EFI=y
 CONFIG_BUILD_NO_GAP_FILL=y
 CONFIG_UART_NS16550_PARENT_INIT_LEVEL=y
+CONFIG_ACPI_DSDT_SUPPORT=y
```
Comment (finikorg) [inline boards/x86/intel_adl/intel_adl_crb_defconfig:16]: The commit message says about "enable uart init as post kernel "

**Comment by jhedberg** [comment]
This PR makes UART work on up_squared, but there are two concerns I have:
- It does this in an indirect way, by disabling `CONFIG_PCIE_PRT`, a Kconfig option that's not mentioned anywhere in the PR or the commit messages. If I understood correctly, it's that option which makes PCIe code depend on ACPI.
- It's not really addressing the main concern, which is that we should work toward getting PCIe and ACPI initialised much earlier in the boot (most likely PRE_KERNEL_1 is the right level)

**Comment by nashif** [comment]
> uart driver is started after uart console and after the logger, this will fail.

actually. one uart is starting in the right level, do we have something not configured with pcie?

**Comment by najumon1980** [comment]
PR heading and content updated

File: lib/acpi/Kconfig [hunk 2]
```diff
@@ -23,7 +23,7 @@ config ACPI_MAX_PRT_ENTRY
 config ACPI_SHELL
 	bool "ACPI command Shell"
 	default y
-	depends on SHELL
+	depends on SHELL && ACPI_DSDT_SUPPORT
```
Comment (dcpleung) [inline lib/acpi/Kconfig:32]: Please provide some reasons in the commit message on why these need to be done.

**Comment by jhedberg** [comment]
@najumon1980 is this PR still relevant? If not, please close it.

**Pull request closed without merging**